invokeai/frontend/web/src/features/gallery/components/Boards/BoardsList/NoBoardBoard.tsx

```diff
@@ -7,31 +7,39 @@ import { defaultSelectorOptions } from 'app/store/util/defaultMemoizeOptions';
 import InvokeAILogoImage from 'assets/images/logo.png';
 import IAIDroppable from 'common/components/IAIDroppable';
 import SelectionOverlay from 'common/components/SelectionOverlay';
-import { boardIdSelected } from 'features/gallery/store/gallerySlice';
+import {
+  boardIdSelected,
+  autoAddBoardIdChanged,
+} from 'features/gallery/store/gallerySlice';
 import { memo, useCallback, useMemo, useState } from 'react';
 import { useBoardName } from 'services/api/hooks/useBoardName';
 import AutoAddIcon from '../AutoAddIcon';
 import BoardContextMenu from '../BoardContextMenu';
+import { useIsReadyToInvoke } from 'common/hooks/useIsReadyToInvoke';
 interface Props {
   isSelected: boolean;
 }
 
 const selector = createSelector(
   stateSelector,
   ({ gallery }) => {
-    const { autoAddBoardId } = gallery;
-    return { autoAddBoardId };
+    const { autoAddBoardId, autoAssignBoardOnClick } = gallery;
+    return { autoAddBoardId, autoAssignBoardOnClick };
   },
   defaultSelectorOptions
 );
 
 const NoBoardBoard = memo(({ isSelected }: Props) => {
   const dispatch = useAppDispatch();
-  const { autoAddBoardId } = useAppSelector(selector);
+  const { autoAddBoardId, autoAssignBoardOnClick } = useAppSelector(selector);
   const boardName = useBoardName(undefined);
+  const isReady = useIsReadyToInvoke();
   const handleSelectBoard = useCallback(() => {
     dispatch(boardIdSelected(undefined));
-  }, [dispatch]);
+    if (autoAssignBoardOnClick && isReady) {
```

Comment:
`useIsReadyToInvoke()` checks more than just if the app is processing. For example, if you are on the img2img tab and you do not have an initial image selected, this will be false.

Instead, `state.system.isProcessing` tells you if the app is mid-generation.